### ISSUE TITLE
Ginkgo CI: validate log gathering functionality for each test upon failure

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#2026]
Parent: #1839 

The currently existing Bash script framework gathers logs when each test finishes. While Ginkgo does not have this functionality (ability to run certain code after all tests in an individual suite finishes), we can add the functionality to gather logs in the `AfterEach` section if a test failures. The functionality should include the following:

- [x] That of `tests/helpers.bash:dump_cli_output`:

Whenever any test fails, the following information should be gathered and accessible for further examination in a PR for both runtime and K8s tests:
```
cilium endpoint list
cilium endpoint get <each endpoint id>
cilium bpf policy list <each endpoint id>
cilium service list
cilium service get <each service id>
cilium identity get <each security identity>
cilium config
cilium bpf lb list
cilium bpf ct list global
cilium bpf tunnel list
cilium policy get
cilium status
```

- [x] That of the stacktrace of the Cilium agent like how it is done in the Bash scripts in `tests/helpers.bash:dump_gops_output`:
```
function dump_gops_output {
  check_num_params "$#" "2"
  local DIR="$1"
  local PROG="$2"
  local PROG_PROFILING_DIR="${DIR}/${PROG}"
  mkdir -p "${PROG_PROFILING_DIR}"
  log "getting gops output for ${PROG} and dumping to dir ${PROG_PROFILING_DIR}"
  local PROG_PID=$(sudo ${GOPS} | grep "${PROG}" | awk '{print $1}')
  log "running \"gops stack\" for ${PROG}"
  sudo ${GOPS} stack ${PROG_PID} > "${PROG_PROFILING_DIR}/${PROG}_stack.txt"
  log "running \"gops memstats\" for ${PROG}"
  sudo ${GOPS} memstats ${PROG_PID} > "${PROG_PROFILING_DIR}/${PROG}_memstats.txt"
  log "running \"gops stats\" for ${PROG}"
  sudo ${GOPS} stats ${PROG_PID} > "${PROG_PROFILING_DIR}/${PROG}_stats.txt"
  log "done getting gops output for ${PROG}"
}
```

- [x] the logs of the KVstore. We should also get the logs of the KVstore, e.g.:
```
docker logs cilium-consul > "${CILIUM_DIR}/consul/consul-logs.txt" 2>/dev/null
```

This should be done in the case of etcd as well.

- [x] contents of `/var/run/cilium` and `/var/lib/cilium` on each VM / in each Cilium Docker Container.

- [x] a dump of all Cilium logs that are dumped to syslog.

**Comment by eloycoto** [comment]
@ianvernon I think that this Issue can be closed. 

Regards

**Comment by ianvernon** [comment]
Merged #2038
